File: Ejercicios_Rubi/Ley_Ohm.rb
# Ley de Ohm
# V=I×R para hayar el voltaje
puts "Ingrese el valor del voltaje del circuito: "
voltaje = gets.chomp.to_f #.to_f para datos flotantes
puts "Ingrese el valor de la resistencia del circuito: "
resistencia = gets.chomp.to_f #.chomp para que el dato ingresado no genere salto de linea al ser utilizado
intensidad = (voltaje/resistencia) #Intesidad entiendase por amperaje
print"Al conectar un resistor de #{resistencia} ohm, a una fuente de #{voltaje} V el voltage circulara una corriente de #{intensidad} amperios"
# V es el voltaje (en voltios, V)
# I es la corriente (en amperios, A)
# R es la resistencia (en ohmios, Ω)

File: Ejercicios Java/Ley_Ohm.java
import java.util.Scanner;

public class Ley_Ohm {
    @SuppressWarnings("ConvertToTryWithResources")
    public static void main(String[] args) {
        // Crear un objeto Scanner para leer las entradas del usuario
        Scanner scanner = new Scanner(System.in);

        // Solicitar el valor del voltaje
        System.out.print("Ingrese el valor del voltaje del circuito: ");
        double voltaje = scanner.nextDouble();

        // Solicitar el valor de la resistencia
        System.out.print("Ingrese el valor de la resistencia del circuito: ");
        double resistencia = scanner.nextDouble();

        // Calcular la intensidad (amperaje)
        double intensidad = voltaje / resistencia;

        // Mostrar el resultado
        System.out.println("Al conectar un resistor de: " + resistencia + " ohm, a una fuente de: " + voltaje + " V, El voltaje circulará una corriente de " + intensidad + " amperios.");

        // Cerrar el scanner
        scanner.close();
    }
}
